Arelle/Arelle
# Add flake8 to CI/CD

#### Reason for change
After the big whitespace cleanup, I figured it'd be a good idea to avoid adding that again. One way of doing that is to use flake8 in the CI/CD.

#### Description of change
This PR:
- Adds flake8 to the dev requirements.
- Runs flake8 as a part of tox.
- Adds flake8 settings that makes it pass for the existing code base.
- Removes trailing new lines for a set of files.

#### Steps to Test
Add whitespace to an arbitrary Python file in the `<repo root>/arelle/` directory and run `flake8 arelle`. You should be seeing this:
```
(.venv) vscode ➜ /workspaces/Arelle (master) $ flake8 arelle
arelle/PackageManager.py:37:41: W291 trailing whitespace
```

My VS code settings cleans up white space automatically so I had to use a text editor to add the white space before testing.

**review**:
@Arelle/arelle
